### ISSUE TITLE
Set non displayed property to null

### ISSUE THREAD

**Issue by mpiot** [SymfonyCasts/dynamic-forms#3]
When the dependent field is not displayed, but is previously set, it should be nice to set it to null or a default value we want.
I don't really know if this can be implemented rirectly, or maybe, mentioned in the documentation to remind people implement it.

Eg:
```php
setMeal(Meal $meal): static {
    $this->meal = $meal;

    if ('breakfast' !== $meal) {
        $this->rating = null;
    }

   return $this;
}
```

**Comment by weaverryan** [comment]
Yes... I haven't really thought about this. I guess, inside the form, if `'breakfast' !== $meal`, the `rating` field will be removed... so the result would be that `setRating()` is never called... and so it would keep whatever value it had when the object was first passed to `->createForm()` (i.e. the value set in the database).

Can you confirm that my statement above is accurate? If so, yes, my feeling is that this would be a good fit for the docs.

**Comment by mpiot** [comment]
Yes, because we remove a field, then we never call the setter and the object keep the initial value. 

I don't really know which is the best way to change that value: 
 - setter
 - change the data when removing or not adding the field
 - add a final `POST_SUBMIT` event that check for example:  if `'breakfast' !=== getMeal()` then `setRating(null)`
